arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -0,0 +1,308 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class CreateSymbolStylesFromWebStylesViewController: UIViewController, UIAdaptivePresentationControllerDelegate {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap(featureLayer: featureLayer)
+            mapView.setViewpoint(
+                AGSViewpoint(
+                    latitude: 34.28301,
+                    longitude: -118.44186,
+                    scale: 1e4
+                )
+            )
+        }
+    }
+    /// The button to show legend table.
+    @IBOutlet var legendBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Instance properties
+    
+    /// The observation on the map view's `mapScale` property.
+    var mapScaleObservation: NSKeyValueObservation?
+    /// The Esri 2D point symbol style created from a web style.
+    let symbolStyle = AGSSymbolStyle(styleName: "Esri2DPointSymbolsStyle", portal: .arcGISOnline(withLoginRequired: false))
+    
+    private var symbolsDataSource: SymbolsDataSource?
+    private var symbols = [(category: SymbolCategory, symbol: AGSSymbol)]() {
+        didSet {
+            symbolsDataSource?.symbols = symbols
+        }
+    }
```

Comment:
Okay, I think I might not have been very clear. If the data used _inside_ the data source (in this case an array of category and symbol pairs) is mutable, then the data source should be prepared to handle changes to that data. However, outside of the data source is a bit of a different story. Here, you know that this property will only be set once after it has been initialized, so this property observer will only be called once. You also know that the table view can only be displayed after this property has been updated, so when this property observer is called, `symbolsDataSource` will be `nil`, since the table view could not have been presented yet. Therefore this property observer isn't needed.

```suggestion
    private var symbols = [(category: SymbolCategory, symbol: AGSSymbol)]()
```

Note: Why treat the data source different? Because the data source doesn't (or shouldn't) know where the data comes from before it gets it. All it knows is that it has a mutable property, so it could be set from the outside world, so it should be prepared to handle changes to it. Does that make sense?

Reply:
Yes that makes sense. It was the comment

> ... right now, if symbols were to change after the table is loaded, nothing would happen. 

that made me a bit confused. Because I thought you were suggesting that 

- the `SymbolsDataSource.symbols` should be mutable, so that whenever `CreateSymbolStylesFromWebStylesViewController.symbols` get updated, the table should also reload, even if the table is being presented
- even if in this sample the `CreateSymbolStylesFromWebStylesViewController.symbols` is only set once, we should still show how to handle async updates as good practices

With your new comments, now my understanding is that

- I'll make the `SymbolsDataSource.symbols` property immutable, and remove its property observers, since it is only initialized when the button is tapped.
- In this sample we don't reload the table when it is being presented

Does it sound correct? I'll make some changes and request your review again... 😅